zookeeper-server/src/main/java/org/apache/zookeeper/server/ServerCnxn.java

```diff
@@ -147,8 +149,9 @@ public void incrOutstandingAndCheckThrottle(RequestHeader h) {
         if (h.getXid() <= 0) {
             return;
         }
-        if (zkServer.shouldThrottle(outstandingCount.incrementAndGet())) {
-            disableRecv(false);
+        if (zkServer.isPresent()
```

Comment:
I am not sure this is what we want,
previously we would have seen an NPE here,
now we aren't doing anything.

This is just an example,
can you please explain better the intent of the patch ?

Reply:
My intent was to remove the NPE.  It is valid that `zkServer` could be `null` and yet there is no check for it (please review the original JIRA).

Is there something else that should be done here?  Comments make it clear that `zkServer` has a valid value of `null`.